src/main/java/org/kcctl/command/ApplyCommand.java

```diff
@@ -68,17 +71,42 @@ public ApplyCommand(ConfigurationContext context) {
     @SuppressWarnings("unchecked")
     @Override
     public Integer call() throws Exception {
+        validate();
         KafkaConnectApi kafkaConnectApi = RestClientBuilder.newBuilder()
                 .baseUri(context.getCurrentContext().getCluster())
                 .build(KafkaConnectApi.class);
+        int returnCode = 0;
+        for (File file : files) {
+            returnCode = max(returnCode, applyOrValidateConnector(kafkaConnectApi, file));
+        }
+        return returnCode;
+    }
+
+    private void validate() {
+        if (files.length > 1 && name != null)
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "It is not possible to use -n with multiple -f. Please provide the connector name in the connector configuration file.");
+    }
 
+    @SuppressWarnings("uncheckd")
+    private int applyOrValidateConnector(KafkaConnectApi kafkaConnectApi, File file) throws Exception {
+        String contents = readConfig(file);
+        Map<String, Object> config = mapper.readValue(contents, Map.class);
+        if (dryRun) {
+            return validateConfigs(kafkaConnectApi, config);
+        }
+        else {
+            return createOrUpdateConnector(kafkaConnectApi, contents, config);
+        }
+    }
+
+    private String readConfig(File file) {
         String contents;
         if (file.getName().equals("-")) {
             contents = readFromStdin();
         }
         else if (!file.exists()) {
-            spec.commandLine().getOut().println("Given file does not exist: " + file.toPath().toAbsolutePath());
-            return 1;
+            throw new CommandLine.ExecutionException(spec.commandLine(), "Given file does not exist: " + file.toPath().toAbsolutePath());
```

Comment:
What's the reason for raising an exception rather than returning a none 0 value?

Reply:
Good question, I asked to myself if it was a good idea.
The pragmatic answer is because the method return a String ^_^
But it can be manage of course
The underneath question is more global. It is about how to manage error when multiple connector are processed.

- Should we run everything and then report all errors at the end ?
- Should we stop on first error

But for the `readConfig` specific case, maybe it would be smarter to first read all the files and then only if evrything is ok, launch the apply process.

What do you thing ?

Reply:
> But for the readConfig specific case, maybe it would be smarter to first read all the files and then only launch the apply process if evrything is ok.

Yes, I think that makes sense. In general, I would

* Try to do as much validation/verification before starting mutating actions
* Abort after the first failure of a mutating action